common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -995,6 +995,15 @@ object CelebornConf extends Logging {
     }.toMap
   }
 
+  def addDeprecatedConfig(entry: ConfigEntry[_], alt: (String, String => String)): Unit = {
```

Comment:
this function is dangerous to expose to the public scope. Currently, it was invoked by `register` and `unregister`, both methods are guarded by `confEntriesUpdateLock`, so it's actually no thread-safe issue. But ideally, `configsWithAlternatives` should have its own guard.

Reply:
How about the current? add `private` to it. and it won't be called in multiple thread.